examples/add_custom_searchbox.html

```diff
@@ -42,33 +42,30 @@
 <script>
   var input;
   function main () {
-    // define map settings
-    var map = L.map('map', {
-      zoomControl: false,
-      center: [41.390205, 2.154007],
-      zoom: 4
-    });
-    // add basemap to our map object
-    L.tileLayer('http://{s}.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png', {attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>'}).addTo(map);
+    
+    var layerUrl = 'https://documentation.carto.com/api/v2/viz/34dfd2e4-d62b-11e5-b5e7-0ea31932ec1d/viz.json';
 
     // add CartoDB layer
-    cartodb.createLayer(map, 'https://documentation.carto.com/api/v2/viz/34dfd2e4-d62b-11e5-b5e7-0ea31932ec1d/viz.json').addTo(map)
-      .done(function () {
-        // When the search button is clicked
-        $('#searchButton').click(function () {
-          // store in the input variable the value from the input box
-          input = $('#ad').val();
-
-          // execute the CARTO SQL API
-          var sql = new cartodb.SQL({ user: 'documentation' });
-          // get the bounds of the geometry which has the same string value (in the name column) than the string value written in the input box
-          sql.getBounds("SELECT * FROM world_borders_1 where name Ilike '" + input + "'").done(function (bounds) {
-            // sets the map view that contains the bounds defined by the getBounds method with the maximum zoom possible
-            map.fitBounds(bounds);
-          });
-        });
-      });
+    cartodb.createVis(map, layerUrl)
+      .done(function (vis,layer) {
+        // set the map options
+        map = vis.getNativeMap().setView([41.390205, 2.154007],4);
+       })
   }
+
+  // When the search button is clicked
+  $('#searchButton').click(function () {
```

Comment:
This needs would need to be moved inside the callback for the `.done`, otherwise it wouldn't work if users try to search before the vis has been "created". Something like:

```
  function main () {
    
    cartodb.createVis('map', 'https://documentation.carto.com/api/v2/viz/34dfd2e4-d62b-11e5-b5e7-0ea31932ec1d/viz.json')
      .done(function (vis,layer) {
        // set the map options
        var map = vis.getNativeMap();
        map.setView([41.390205, 2.154007],4);

        // When the search button is clicked
        $('#searchButton').click(function () {
          // store in the input variable the value from the input box
          var query = $('#ad').val();

          // execute the CARTO SQL API
          var sql = new cartodb.SQL({ user: 'documentation' });

          // get the bounds of the geometry which has the same string value (in the name column) than the string value written in the input box
          sql.getBounds("SELECT * FROM world_borders_1 where name ILIKE '" + query + "'").done(function (bounds) {
            // sets the map view that contains the bounds defined by the getBounds method with the maximum zoom possible
            map.fitBounds(bounds);
          });
        });
      })
  }
```